### DISCUSSION TITLE
Reload Relation Manager after an Action

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3902] [Help] [answered]
How can I make a relation manager reload after an action has completed on an edit or view page?

**Comment by danharrin** [comment]
You should emit a Livewire event from a hook, and then listen for it on the relation manager and call `$refresh`.

**Comment by joshembling** [comment, marked as answer]
For anyone interested you can do something like this: 

```
Action:make('my_action')
->after(function ($livewire) {
    $livewire->dispatch('refreshRelation');
}),
```

```
// your relation manager... 

class YourRelationManager extends RelationManager
{
protected $listeners = ['refreshRelation' => '$refresh'];

...
}
```

**Reply by MirkoDzudzarLF** [reply]
I am encountering an issue where a relation manager is not loaded on the screen initially. When I dispatch a job and attempt to reload the relation manager after performing an action, I receive a server error. This issue only occurs when the relation manager has not been previously loaded. If the relation manager has been loaded beforehand, the process works correctly.

"Typed property Filament\Resources\RelationManagers\RelationManager::$table must not be accessed before initialization..."

**Comment by mahemmati** [comment]
I did exactly the same and it works, but "sometimes" I get this exception:

// Typed property Filament\Resources\RelationManagers\RelationManager::$table must not be accessed before initialization

Any ideas?

> For anyone interested you can do something like this:
> 
> ```
> Action:make('my_action')
> ->after(function ($livewire) {
>     $livewire->dispatch('refreshRelation');
> }),
> ```
> 
> ```
> // your relation manager... 
> 
> class YourRelationManager extends RelationManager
> {
> protected $listeners = ['refreshRelation' => '$refresh'];
> 
> ...
> }
> ```

**Reply by jerry-whoami** [reply]
It's cause the table has not been loaded yet. Try adding deferLoading(false)
```php
public function table(Table $table): Table
{
  return $table
    ->deferLoading(false);
}
```

**Reply by mahemmati** [reply]
deferLoading() did not work out.
But setting the $isLazy prop of the relation manager to false solved the issue.
Thanks for the hint.
```php
class ApprovalsRelationManager extends RelationManager
{
    protected static bool $isLazy = false;
}
```

**Comment by howdu** [comment]
Not ideal but a way around this 
```php
#[On('refreshRelation')]
public function refreshRelationTable(): void
{
    $rp = new \ReflectionProperty(self::class, 'table');
    if (! $rp->isInitialized($this)) {
        parent::bootedInteractsWithTable();
    }
}
```

**Reply by MirkoDzudzarLF** [reply]
Thank you for your suggestion. In my case, I believe a more suitable approach would be to perform a full page refresh with a redirect, as follows:
`return redirect(ViewUser::getUrl(['record' => $record->id]));`

I appreciate your help and will try your suggested method as well. Thank you!
